### ISSUE TITLE
Edit x/^/ i:// : Unexpectedly exuberant

### ISSUE THREAD

**Issue by rjkroege** [rjkroege/edwood#39]
Middling `Edit x/^/ i:// :` inserts `//` before every rune. Conceivably this is the *right* behaviour (is it?) but it deviates significantly from Acme.

**Comment by paul-lalonde** [comment]
It also comes with a nifty selection rendering artifact:
![image](https://user-images.githubusercontent.com/1304784/38471567-f01b12cc-3b40-11e8-9cae-0c1722de21f5.png)

**Comment by paul-lalonde** [comment]
And it would be difficult for me to argue that this is the correct behavior for '^'

**Comment by rjkroege** [comment]
Hm. I didn't get the background colours. But indeed, that's unfortunate.

**Comment by paul-lalonde** [comment]
This is unexpectedly ugly.  Instead of porting the acme regexp, I used Go's builtin, and it matches up reasonably well.  Until you want multiple "start of line" chars from a Reader.  When you re-start, you get the "start of selection" match for ^, instead of the start of line.  One solution is to find all of them, but the go regexp library offers FindReaderSubmatchIndex but not the FindAll variant.

**Comment by rjkroege** [comment]
I think that it could be lived with? I think that I usually use the `^` on whole lines only so would seem fine?

**Comment by paul-lalonde** [comment]
Even more fun: acme itself is broken on $ matches.  Select the middle line of
```
this is a
short text
to try Edit
```
and execute Edit x/$/a/junk/
Receive
```
this is ajunk
short text
to try Edit
```
rather than the expected
```
this is a
short textjunk
to try Edit
```

Edwood will get this right, and not match ^ and $ to start and end of range.  Both will match start and end of document.

**Comment by paul-lalonde** [comment]
Fixed in 03f4790ff10a6fba000a65df2adb4380924e8671

Backwards searches remain broken.
